weave/tests/trace/test_weave_client.py

```diff
@@ -1183,6 +1183,24 @@ def test_table_partitioning(network_proxy_client):
     )
 
 
+def test_call_batch_size_handling(network_proxy_client):
+    client, remote_client, records = network_proxy_client
+
+    @weave.op
+    def save_large_string(len: int):
+        return "x" * len
+
+    @weave.op
+    def save_large_dict(len: int):
+        return {f"{i}": "x" for i in range(len)}
+
+    save_large_string(10_000_000)
+    save_large_dict(10_000_000)
+
+    calls = client.get_calls()
+    assert len(calls) == 2
```

Comment:
probably want to add more assertions here to ensure that you logic works. As implemented, i don't think your redaction code is actually running